src/services/studio/studio.service.ts

```diff
@@ -181,24 +180,69 @@ class StudioService {
         return variablesByKey;
     }
 
-    private async movePackageToSpace(manifest: StudioPackageManifest): Promise<void> {
-        const nodeInTargetTeam = await nodeApi.findOneByKeyAndRootNodeKey(manifest.packageKey, manifest.packageKey);
+    public async mapSpaces(exportedFiles: AdmZip, studioManifests: StudioPackageManifest[]): Promise<AdmZip> {
+            for (const packageZipFile of exportedFiles.getEntries()) {
+                if(packageZipFile.name.endsWith(".zip")) {
+                    const packageKey = this.getPackageKeyAndVersion(packageZipFile.name).packageKey;
+                    if(this.isStudioPackage(studioManifests, packageKey)) {
+                        const studioManifest = studioManifests.find(manifest => manifest.packageKey === packageKey);
+                        if (studioManifest) {
+
+                            const spaceId = await this.findDesiredSpaceIdForPackage(packageKey, studioManifest);
+                            studioManifest.space.id = spaceId;
+
+                            const packageZip = new AdmZip(packageZipFile.getData());
+                            packageZip.getEntries().forEach(nodeFile => {
+                                if (nodeFile.entryName.endsWith("yml")) {
```

Comment:
Do we have constants for these filename extensions? Same for `.zip`

Reply:
added them.

Reply:
Did some changes with the latest commits, can you re-review?